cstore_writer.c

```diff
@@ -1033,3 +1003,21 @@ SyncAndCloseFile(FILE *file)
 						errmsg("could not close file: %m")));
 	}
 }
+
+
+/* CopyStringInfo creates a deep copy of given source string */
+StringInfo
+CopyStringInfo(StringInfo sourceString)
```

Comment:
I wanted to use as little space as needed. appendStringInfo tries to make sure it has enough space by calling enlargeStringInfo, which allocates as twice as memory needed trying to save from additional allocations that could be requested later.  so this function is not to save from line of code. it is only meant to allocate right amount of memory.